vm/actor/tests/deadline_test.rs

```diff
@@ -0,0 +1,671 @@
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 2, &[4]).unwrap();
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 3, &[0]).unwrap();
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 3, &[3]).unwrap();
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 3, &[5]).unwrap();
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 4, &[5]).unwrap();
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 5, &[0]).unwrap();
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 5, &[7]).unwrap();
+}
+
+const NEW_SECTOR_PART_SIZE: usize = 4;
+
+#[test]
+fn assign_new_sectors_test() {
+    let mut deadlines = assign_sectors_setup(Deadlines::new(), &seq(0, 0));
+    DeadlineBuilder::new(&[]).verify(&deadlines);
+
+    deadlines = assign_sectors_setup(Deadlines::new(), &seq(0, 1));
+    DeadlineBuilder::new(&[0, 1]).verify(&deadlines);
+
+    deadlines = assign_sectors_setup(Deadlines::new(), &seq(0, 15));
+    DeadlineBuilder::new(&[0, 4, 4, 4, 3]).verify(&deadlines);
+
+    deadlines = assign_sectors_setup(
+        Deadlines::new(),
+        &seq(0, (WPOST_PERIOD_DEADLINES - 1) * NEW_SECTOR_PART_SIZE + 1),
+    );
+    DeadlineBuilder::new(&[])
+        .add_to_all_from(1, NEW_SECTOR_PART_SIZE)
+        .add_to(1, 1)
+        .verify(&deadlines);
+}
+
+#[test]
+fn incremental_assignment_test() {
+    // Add one sector at a time.
+    let mut deadlines = DeadlineBuilder::new(&[0, 1]).deadlines;
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(1, 1));
+    DeadlineBuilder::new(&[0, 2]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(2, 1));
+    DeadlineBuilder::new(&[0, 3]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(3, 1));
+    DeadlineBuilder::new(&[0, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(4, 1));
+    DeadlineBuilder::new(&[0, 4, 1]).verify(&deadlines);
+
+    // Add one partition at a time.
+    deadlines = Deadlines::new();
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(0, 4));
+    DeadlineBuilder::new(&[0, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(4, 4));
+    DeadlineBuilder::new(&[0, 4, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(2 * 4, 4));
+    DeadlineBuilder::new(&[0, 4, 4, 4]).verify(&deadlines);
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(3 * 4, 4));
+    DeadlineBuilder::new(&[0, 4, 4, 4, 4]).verify(&deadlines);
+    // Add lots
+    deadlines = Deadlines::new();
+    deadlines = assign_sectors_setup(deadlines.clone(), &seq(0, 2 * NEW_SECTOR_PART_SIZE + 1));
+    DeadlineBuilder::new(&[
+        0,
+        NEW_SECTOR_PART_SIZE as u64,
+        NEW_SECTOR_PART_SIZE as u64,
+        1,
+    ])
+    .verify(&deadlines);
+    deadlines = assign_sectors_setup(
+        deadlines.clone(),
```

Comment:
```suggestion
    deadlines = assign_sectors_setup(deadlines, &seq(1, 1));
    DeadlineBuilder::new(&[0, 2]).verify(&deadlines);
    deadlines = assign_sectors_setup(deadlines, &seq(2, 1));
    DeadlineBuilder::new(&[0, 3]).verify(&deadlines);
    deadlines = assign_sectors_setup(deadlines, &seq(3, 1));
    DeadlineBuilder::new(&[0, 4]).verify(&deadlines);
    deadlines = assign_sectors_setup(deadlines, &seq(4, 1));
    DeadlineBuilder::new(&[0, 4, 1]).verify(&deadlines);

    // Add one partition at a time.
    deadlines = Deadlines::new();
    deadlines = assign_sectors_setup(deadlines, &seq(0, 4));
    DeadlineBuilder::new(&[0, 4]).verify(&deadlines);
    deadlines = assign_sectors_setup(deadlines, &seq(4, 4));
    DeadlineBuilder::new(&[0, 4, 4]).verify(&deadlines);
    deadlines = assign_sectors_setup(deadlines, &seq(2 * 4, 4));
    DeadlineBuilder::new(&[0, 4, 4, 4]).verify(&deadlines);
    deadlines = assign_sectors_setup(deadlines, &seq(3 * 4, 4));
    DeadlineBuilder::new(&[0, 4, 4, 4, 4]).verify(&deadlines);
    // Add lots
    deadlines = Deadlines::new();
    deadlines = assign_sectors_setup(deadlines, &seq(0, 2 * NEW_SECTOR_PART_SIZE + 1));
    DeadlineBuilder::new(&[
        0,
        NEW_SECTOR_PART_SIZE as u64,
        NEW_SECTOR_PART_SIZE as u64,
        1,
    ])
    .verify(&deadlines);
    deadlines = assign_sectors_setup(
        deadlines,
```
Clones are not needed, the test could be setup differently, but this is quickest solution